core/src/scene/sceneLoader.cpp

```diff
@@ -1800,6 +1800,12 @@ void SceneLoader::loadLayer(const std::pair<Node, Node>& layer, const std::share
     }
 
     auto sublayer = loadSublayer(layer.second, name, scene);
+    if (!sublayer.enabled()) {
+        auto dataSource = scene->getTileSource(source);
+        if (dataSource) {
+            dataSource->generateGeometry(dataSource->generateGeometry() | false);
```

Comment:
This doesn't make sense to me. Did you mean to use the bitwise OR here? Isn't `(x OR false) == x`?

Reply:
oops vim messed up. "OR", will update.

Reply:
`(x OR false) == x`

Is what I am looking for. `generateGeometry` is set to `false` before. But if there are multiple layers in the scene using a datasource, `generateGeometry` to be set to `false` only if all these layers are disabled.

Reply:
`dataSource->generateGeometry(dataSource->generateGeometry() | false);`
Is equivalent to:
`dataSource->generateGeometry(dataSource->generateGeometry());`
I think something is missing here.

Reply:
I want it to be a `||`. Was a typo.